### PR TITLE
fix: predicate estimation for inputs with predicate field

### PR DESCRIPTION
**Pull request by luizstacio**
close: #1129 

In this PR;
- Change check to assign predicate gas to use the `predicate` field as rule
- Add tests with multiple inputs with predicate to ensure gas is correctly assign

### CODE REVIEW

**Review by Torres-ssf — DISMISSED** [review]

**Review by Torres-ssf — DISMISSED** [review]

**Review by Torres-ssf — DISMISSED** [review]

**Review by danielbate — APPROVED** [review]

File: packages/providers/src/provider.ts [hunk 1]
```diff
@@ -398,7 +398,7 @@ export default class Provider {
 
     if (decodedTransaction.inputs) {
       decodedTransaction.inputs.forEach((input, index) => {
-        if (input.type === InputType.Coin && input.predicate) {
+        if ('predicate' in input && input.predicateGasUsed.gt(0)) {
```
[the remark this answers is not in the document]
Comment (luizstacio) [reply packages/providers/src/provider.ts:401]: But why would we ignore the predicate message? If they are a predicate they need the gas to be assigned

File: packages/wallet/src/test-utils/seedTestWallet.ts [hunk 2]
```diff
@@ -2,9 +2,10 @@ import { randomBytes } from '@fuel-ts/keystore';
 import { coinQuantityfy, ScriptTransactionRequest } from '@fuel-ts/providers';
 import type { CoinQuantityLike } from '@fuel-ts/providers';
 
+import type { Account } from '../account';
 import { WalletUnlocked } from '../wallets';
 
-export const seedTestWallet = async (wallet: WalletUnlocked, quantities: CoinQuantityLike[]) => {
+export const seedTestWallet = async (wallet: Account, quantities: CoinQuantityLike[]) => {
```
[the remark this answers is not in the document]
Comment (luizstacio) [reply packages/wallet/src/test-utils/seedTestWallet.ts:8]: In this way the seedTestWallet will work with any Account type, like predicate

**Review by Torres-ssf — APPROVED** [review]

File: packages/fuel-core/VERSION [hunk 3]
```diff
@@ -1 +1 @@
-0.19.0
+0.19.1
```
Comment (Dhaiwat10) [inline packages/fuel-core/VERSION:1]: Do we need this version bump for the changes in this PR to work or is it unrelated?

**Review by Dhaiwat10 — APPROVED** [review]

File: packages/fuel-core/VERSION (hunk 3 above)
Comment (luizstacio) [reply packages/fuel-core/VERSION:1]: Is not completely related but it has a fix for GTF message data.

**Pull request merged**